src/v/storage/log_manager.cc

```diff
@@ -142,12 +142,12 @@ log_manager::log_manager(
   , _kvstore(kvstore)
```

Comment:
> storage: split housekeeping/gc utilities in log_manager
> To be specific about the future uses of the semaphore and jitter variables,
> rename them to `_housekeeping_jitter`. Also, add a new `_gc_sem`.

it seems like this commit claims to do 3 things, where as it does 2? I don't know what "split utilities" really means, but I only see a rename and the introduction of gc_sem. 

A commit should ideally just do one thing.